### PR TITLE
Support for Json web tokens implemented

### PR DESCRIPTION
**Pull request by jszymanski97**
New UserLogin Request and Response DTO created
New AuthenticationService, JwtAuthenticationFilter and JwtUtil classes added

### CODE REVIEW

**Comment by jszymanski97** [comment]
Not sure why but I had to comment my @Value("${jwt.expiration}") as mvn clean was keep throwing errors. However I was testing the app by running it without hard coding the values and it worked fine...

File: src/main/java/mate/project/controller/AuthenticationController.java [hunk 1]
```diff
@@ -15,9 +18,15 @@
...
     private final UserService userService;
+    private final AuthenticationService authenticationService;
 
     @PostMapping("/registration")
     public UserResponseDto register(@Valid @RequestBody UserRegistrationRequestDto request) {
         return userService.register(request);
     }
+
+    @PostMapping("/login")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto request) {
```
Comment (MateuszRostek) [inline src/main/java/mate/project/controller/AuthenticationController.java]: You should also add validation, so:
```suggestion
    public UserLoginResponseDto login(@RequestBody @Valid UserLoginRequestDto request) {
```

File: src/main/java/mate/project/security/AuthenticationService.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationService {
+    private final JwtUtil jwtUtil;
+    private final AuthenticationManager authenticationManager;
+
+    public UserLoginResponseDto authenticate(UserLoginRequestDto request) {
+        Authentication authenticate = authenticationManager.authenticate(
```
Comment (MateuszRostek) [inline src/main/java/mate/project/security/AuthenticationService.java]: More clear name:

```suggestion
        Authentication authentication = authenticationManager.authenticate(
```

File: src/main/java/mate/project/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
```
Comment (MateuszRostek) [inline src/main/java/mate/project/security/JwtAuthenticationFilter.java:24]: Redundant empty line

File: src/main/java/mate/project/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String token = request.getHeader("Authorization");
+        if (StringUtils.hasText(token) && token.startsWith("Bearer ")) {
+            return token.substring(7);
```
Comment (MateuszRostek) [inline src/main/java/mate/project/security/JwtAuthenticationFilter.java]: Please make all magic numbers/string literals constants.

**Review by MateuszRostek — CHANGES_REQUESTED** [review]
Nice try! Please look at my comments and implement them.

Also, if you are having troubles with something (like here with running `mvn clean package`) ask on the chat on our platform instead of sending incorrect solution to review. Please add your question on the platform chat along with a screenshot of the error you are receiving.

**Comment by jszymanski97** [comment]
I have commited my changes but they failed checks as I made mistake in test/resources but have fixed it with second commit.

File: pom.xml [hunk 5]
```diff
@@ -6,7 +6,6 @@
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
         <version>3.3.3</version>
-        <relativePath/>
```
Comment (MateuszRostek) [inline pom.xml]: Why did you remove this? You should change it back.

File: pom.xml [hunk 6]
```diff
@@ -32,6 +31,8 @@
         <lombok.version>1.18.34</lombok.version>
         <lombok.mapstruct.binding.version>0.2.0</lombok.mapstruct.binding.version>
         <mapstruct.version>1.5.5.Final</mapstruct.version>
+        <jjwt.version>0.11.5</jjwt.version>
+        <spring.config.location>src/main/resources/application.properties</spring.config.location>
```
Comment (MateuszRostek) [inline pom.xml]: This is also redundant

File: pom.xml [hunk 7]
```diff
@@ -50,6 +51,11 @@
             <artifactId>spring-boot-starter-data-jpa</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-configuration-processor</artifactId>
+        </dependency>
```
Comment (MateuszRostek) [inline pom.xml]: This is also not needed to do this task

File: src/main/java/mate/project/dto/UserLoginRequestDto.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package mate.project.dto;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.Size;
+
+public record UserLoginRequestDto(
+        @NotEmpty
```
Comment (MateuszRostek) [inline src/main/java/mate/project/dto/UserLoginRequestDto.java]: Not blank is beter here
```suggestion
        @NotBlank
```

File: src/main/java/mate/project/dto/UserLoginRequestDto.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+        @Email
+        @Size(min = 8, max = 255)
+        String email,
+        @NotEmpty
```
Comment (MateuszRostek) [inline src/main/java/mate/project/dto/UserLoginRequestDto.java]: same here
```suggestion
        @NotBlank
```

**Review by MateuszRostek — APPROVED** [review]
Good job overall! Check comments and implement them, approved!

File: pom.xml (hunk 5 above)
Comment (jszymanski97) [reply pom.xml]: @MateuszRostek Sorry this was one of many steps I took trying to solve the issues I had with mvn clean package not passing without giving me too much info on what was wrong... But all sorted. Will correct this
